crepecake/src/test/java/com/google/cloud/tools/crepecake/image/LayerBuilderTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.image;
+
+import com.google.cloud.tools.crepecake.blob.Blob;
+import com.google.cloud.tools.crepecake.tar.TarStreamBuilder;
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+/** Tests for {@link LayerBuilder}. */
+public class LayerBuilderTest {
+
+  @Mock private Blob mockBlob;
+  @Mock private TarStreamBuilder mockTarStreamBuilder;
+
+  @Before
+  public void setUpMocks() throws IOException {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @Test
+  public void testBuild() throws LayerPropertyNotFoundException {
+    Mockito.when(mockTarStreamBuilder.toBlob()).thenReturn(mockBlob);
+
+    // Fake files to build into the layer.
+    List<TarArchiveEntry> fileEntries =
+        Arrays.asList(
+            new TarArchiveEntry(new File("fileA"), "/path/to/fileA"),
+            new TarArchiveEntry(new File("directory/fileB"), "/path/to/directory/fileB"),
+            new TarArchiveEntry(new File("directory/fileC"), "/path/to/directory/fileC"),
+            new TarArchiveEntry(new File("directory/"), "/path/to/directory/"));
+
+    LayerBuilder layerBuilder = new LayerBuilder(() -> mockTarStreamBuilder);
```

Comment:
I think if you democked this the test would be more useful. We can certainly write a real tar file during a test.

Reply:
Since the TarStreamBuilderTest already verifies that the TarStreamBuilder builds the right tar, this test was intended to just test that the file entries were added to the TarStreamBuilder correctly. I'll remove all of the Supplier and API-changing stuff.